src/DataStructures/TaggedVariant.hpp

```diff
@@ -136,6 +141,15 @@ class TaggedVariant {
 
   void pup(PUP::er& p) { p | data_; }
 
+  /// A TaggedVariant over option tags can be parsed as any of them.
+  /// @{
+  static constexpr Options::String help = "One of multiple options";
```

Comment:
Is there a way that we can specify this string? Possibly from a template parameter somehow?

I've tested this PR out on some domain creator FoT changes I'm making, and it works really well! I think this will make input files much clearer. The only thing is that when a typo is made in the input file, this help string isn't very helpful. E.g. this was (part of) the error message I got

```
In string:
  While parsing option TestCreationOpt:
  While creating a TranslationMap:
  While parsing option InitialValues:
  While creating a TaggedVariant:
  At line 3 column 5:
  Cannot decide between alternative options.
  
==== Parsing the option string:
  FromVolFil:
    H5Filename: BbhVolume0.h5
    SubfileName: VolumeData
    Time: 0.0
  
==== Description of expected options:
  One of multiple options
  
Options:
    EITHER
      Explicit:
        Something useful here
  
  OR
      FromVolFile:
        Something useful here

```

If under the `Description of expected options` part I was able to use the help string for the actual option this was creating, that'd be much clearer I think (i.e. `Initial values for the translation map, its velocity and acceleration.`)

Reply:
There's nowhere to put a template parameter, unfortunately.

What might be good in general is if there was an error when creating a class it also displayed the help from the tag, but currently the code generating the error message doesn't know that information.

We could add the help test for each option after the "While parsing option" lines in the "backtrace" pretty easily.  It's not the most visible place, but at least it would be there.  Example from ExportCoordinates:
```
In /home/wthrowe/spectre/tests/InputFiles/ExportCoordinates/Input1D.yaml
While parsing option DomainCreator
  The domain to create initially
While creating a unique_ptr
While operating factory for DomainCreator
While creating a Interval
While parsing option TimeDependence
  The time dependence of the moving mesh domain.
While creating a unique_ptr
While operating factory for TimeDependence
While creating a UniformTranslation
At line 48 column 9:
Option 'VelocityX' is not a valid option.

==== Parsing the option string:
InitialTime: 0.0
VelocityX: [0.5]

==== Description of expected options:
A spatially uniform translation initialized with a constant velocity.

Options:
  InitialTime:
    type=double
    The initial time of the functions of time

  Velocity:
    type=[double x1]
    The velocity of the map.
```

Reply:
@wthrowe I'm ok forgoing this change for the time being. I think the benefit to the input file syntax is enough for the moment.